spec/support/redis_multiplexer.rb

```diff
@@ -87,7 +87,7 @@ def equalish?(a, b)
     elsif a.is_a?(Array) && b.is_a?(Array)
       a.collect { |c| masked(c.to_s) }.sort == b.collect { |c| masked(c.to_s) }.sort
     elsif a.is_a?(Exception) && b.is_a?(Exception)
-      a.class == b.class && a.message == b.message
+      a.class == b.class && a.message.split("\n", 2).first == b.message.split("\n", 2).first
```

Comment:
Resolve issues running the test suite on Ruby 3.1.

The error message has been extended is some circumstances to point to positions in the codebase. This exposes differences in implementation between the real Redis and the mock one:

```diff
undefined method `to_a' for 1:Integer

<     hmset(key, hash.to_a.flatten)
<                    ^^^^^
>       kvpairs = hash.to_a.flatten
>                     ^^^^^
Did you mean?  to_c
               to_r
               to_f
               to_i
               to_s
```

I propose reducing the comparison to just the first line of the error message.